### PR TITLE
Handle special cases of trigonometric functions.

### PR DESCRIPTION
**Pull request by jsuyash1514**
This PR adds the results for the corner cases of trigonometric functions sin, cos, and tan.
Solves #1816 
@ewpatton Please have a look.

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by jsuyash1514** [comment]
@ewpatton I've added some negative values in the tests. Please have a look.

**Comment by halatmit** [comment]
I'm not following this carefully, but it seems that the tests are testing
only integer arguments.

Are we also testing that identities are obeyed, e.g., is tan(x)*cos(x)=
sin(x) for all x?


== Hal

Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by jsuyash1514** [comment]
@ewpatton @halatmit I've added a test for a trigonometric identity tan(x)*cos(x) = sin(x). Please let me know if the way I've implemented this test is correct. If so, I would like to add some more trigonometric identities and equations.

File: appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java [hunk 1]
```diff
@@ -1123,6 +1142,24 @@ public void testAtan() throws Throwable {
      testUnaryDoubleFunction("atan-degrees", args, vals);
    }
 
+   public void testTrigonometricIdentities() throws Throwable {
+       Random random = new Random();
```
Comment (ewpatton) [inline appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java]: Indentation is off here. Should be 4 spaces, not 6.

File: appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java [hunk 2]
```diff
@@ -13,6 +13,9 @@
 import junit.framework.TestCase;
 import kawa.standard.Scheme;
 
+import java.util.Random;
+
+
```
Comment (ewpatton) [inline appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java]: Google style guide has the imports placed alphabetically, so the java import should come after gnu and before junit. You also don't need the extra blank lines.

File: appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm [hunk 3]
```diff
@@ -1793,13 +1793,31 @@
           360))
 
 (define (sin-degrees degrees)
-  (sin (degrees->radians-internal degrees)))
+  (if(= (modulo degrees 90) 0)
```
Comment (ewpatton) [inline appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm]: Minor code style thing: I think generally it's accepted to put a space before the first `(` in a sequence of `(`, so there should be a space between the `if` and the `(` that comes after it, e.g., `(if (=`. I recommend this change for everywhere it occurs in your change.

**Comment by jsuyash1514** [comment]
@ewpatton I've made the requested changes and resolved the conflicts with the base branch. Kindly review.

**Comment by jsuyash1514** [comment]
@ewpatton I've added tests for some more trigonometric identities and equations. Please have a look and let me know if anything remains to be done.

**Pull request merged**